lib/class-wp-rest-widgets-controller.php

```diff
@@ -116,7 +92,7 @@ public function register_routes() {
 			),
 			'schema' => array( $this, 'get_public_item_schema' ),
 		) );
-		register_rest_route( $this->namespace, '/' . $this->rest_base .'/types/(?P<type>[\w-]+)', array(
+		register_rest_route( $this->namespace, '/' . $this->rest_base_types . '/(?P<type>[\w-]+)', array(
```

Comment:
> In other words, I'm realizing that maybe widget-types is not needed at all since it could be made available via a schema request. What I'm not clear on is how to deal with including multiple widget types in the schema for a single endpoint. Perhaps the posts endpoint has that laid out for us.

On individual endpoints, an `OPTIONS` request will return the schema for the resource.

Because different widgets have different schemas, they'd be expected to be exposed at different routes. You'll notice we have `/wp/v2/posts`, `/wp/v2/pages` and `/wp/v2/media`, all because the data is modeled differently.